### PR TITLE
Enable automatic update on create as well

### PR DESCRIPTION
**Pull request by jproyo**

### CODE REVIEW

**Review by loony-bean — APPROVED** [review]

File: v1/node.proto [hunk 1]
```diff
@@ -230,6 +230,8 @@ message CreateNodeRequest {
   repeated FilteredIpAddr allow_ips = 9;
   // A list of ip addresses denied all access to any ports on this node.
   repeated FilteredIpAddr deny_ips = 10;
+  // Whether or not to enable automatic update on the node.
+  optional bool self_update = 11;
```
Comment (loony-bean) [inline v1/node.proto:234]: why optional?

File: v1/node.proto [hunk 2]
```diff
@@ -142,8 +143,30 @@ message Node {
...
+  // Whether or not the node should upgrade itself.
+  bool enabled = 1;
+  // The type of upgrade that is being performed.
+  UpgradePolicy policy = 2;
+  // The type of upgrade that is being performed.
+  enum UpgradePolicy {
+    // Force Upgrade all versions no matter what
+    UPGRADE_POLICY_UNSPECIFIED = 0;
+    // Upgrade all versions except major versions.
+    UPGRADE_POLICY_NOT_MAJOR = 1;
```
Comment (loony-bean) [inline v1/node.proto:165]: do we also need something like UPGRADE_POLICY_ALL?

File: v1/node.proto [hunk 3]
```diff
@@ -142,8 +143,30 @@ message Node {
...
+// to confirm the upgrade.
+message SelfUpgrade {
+  // Whether or not the node should upgrade itself.
+  bool enabled = 1;
+  // The type of upgrade that is being performed.
+  UpgradePolicy policy = 2;
+  // The type of upgrade that is being performed.
+  enum UpgradePolicy {
+    // Force Upgrade all versions no matter what
+    UPGRADE_POLICY_UNSPECIFIED = 0;
```
Comment (loony-bean) [inline v1/node.proto:163]: i believe we cannot use UPGRADE_POLICY_UNSPECIFIED for that

File: v1/command.proto [hunk 4]
```diff
@@ -154,7 +154,8 @@ message NodeUpgrade {
 }
 
 message NodeUpdate {
-  optional bool self_update = 1;
+  // not needed anymore
+  //optional bool self_update = 1;
```
Comment (loony-bean) [inline v1/command.proto:158]: btw imho we can just remove the commented items and reindex
we do not need to be so strict now

File: v1/node.proto (hunk 2 above)
Comment (jproyo) [reply v1/node.proto:165]: I am thinking the same. As far as the code base is setup it seems that `_UNSPECIFIED` fields are taken as defaults. So, we can explicitly add an `ALL` or just map `UNSPECIFIED` as `All` like the default.
Perhaps the former is more verbose but more expressive.

**Comment by jproyo** [comment]
I am closing this PR, because we need to get back to `bool self_update`, as per @densone recommendation we are going with a all or nothing approach on automatic update.

**Pull request closed without merging**